src/common/webrequest_curl.cpp

```diff
@@ -356,16 +369,511 @@ void wxWebAuthChallengeCURL::SetCredentials(const wxWebCredentials& cred)
     m_request.StartRequest();
 }
 
+
+wxDECLARE_EVENT(wxSocketAction,wxThreadEvent);
+
+class SocketPoller: public wxThreadHelper
+{
+public:
+    enum PollAction
+    {
+        InvalidAction = 0,
+        PollForRead = 1,
+        PollForWrite = 2,
+        PollForError = 4
+    };
+
+    enum Result
+    {
+        InvalidResult= 0,
+        ReadyForRead = 1,
+        ReadyForWrite = 2,
+        HasError = 4
+    };
+
+    SocketPoller();
+    ~SocketPoller();
+    bool StartPolling(wxSOCKET_T, int, wxEvtHandler*);
+    void StopPolling(wxSOCKET_T);
+    void ResumePolling(wxSOCKET_T);
+
+private:
+    enum
+    {
+#ifdef INVALID_SOCKET
+        SOCKET_POLLER_INVALID_SOCKET = INVALID_SOCKET
+#else
+        SOCKET_POLLER_INVALID_SOCKET = ((wxSOCKET_T)(~0))
+#endif
+    };
+
+    class Message
+    {
+        public:
+            enum MessageId
+            {
+                AddSocketAction,
+                DeleteSocketAction,
+                ResumePolling,
+                Quit,
+                LastMessageId
+            };
+
+            Message(MessageId i = LastMessageId,
```

Comment:
```suggestion
            explicit Message(MessageId i = LastMessageId,
```

Reply:
changed in [253c617](https://github.com/wxWidgets/wxWidgets/pull/2193/commits/253c617cafafe727ceec73ac9977ebc9439dee2c)